### PR TITLE
update to Next 14 & Node 20, other tweaks and cleanups

### PR DESCRIPTION
**Pull request by Simek**
# 📝 Why & how

Update to Next 14 & Node 20, add `eslint-config-next`, fix React Native deprecations (in own code), improve a11y, bump few minor dependencies, tweaks & cleanups.

# ✅ Checklist

- [ ] Added library to **`react-native-libraries.json`**
- [ ] Updated library in **`react-native-libraries.json`**

- [ ] Documented in this PR how to use the feature or replicate the bug.
- [ ] Documented in this PR how you fixed or created the feature.

### CODE REVIEW

**Comment by Simek** [comment]
Deployment preview:
* https://react-native-directory-mlz9gjrfs-rndir.vercel.app/

File: pages/_app.tsx [hunk 1]
```diff
@@ -53,6 +53,7 @@ const App = ({ pageProps, Component }) => (
           <Header />
           <Component {...pageProps} />
           <Footer />
+          <Analytics />
```
Comment (brentvatne) [inline pages/_app.tsx:56]: neat

File: package.json [hunk 2]
```diff
@@ -66,10 +67,6 @@
     "typescript": "^5.2.2",
     "yarn-deduplicate": "^6.0.2"
   },
-  "resolutions": {
-    "metro": "^0.73.7",
-    "metro-resolver": "^0.73.7"
-  },
```
Comment (brentvatne) [inline package.json:72]: yay

**Review by brentvatne — APPROVED** [review]
just did a quick pass but looks good, thanks for staying on top of this!

**Pull request merged**